client/core/trade.go

```diff
@@ -1739,79 +1758,112 @@ func (t *trackedTrade) refundMatches(matches []*matchTracker) (uint64, error) {
+			// The server's response timeout may have just passed, but we got
+			// what we needed to do our swap or redeem if the match is still
+			// live, so do not log this as an error.
+		}
+	}()
 
-	// Respond to DEX.
-	err = t.dc.ack(msgID, match.id, audit)
-	if err != nil {
-		return errs.add("Audit error: %v", err)
-	}
 	return nil
 }
 
 // auditContract audits the contract for the match and relevant MatchProof
-// fields are set. The changes are not saved to the database. auditContract must
-// be called with the trackedTrade.mtx locked.
-func (t *trackedTrade) auditContract(match *matchTracker, coinID []byte, contract []byte) error {
+// fields are set. The changes are not saved to the database. This may block for
+// a long period, and should be run in a goroutine. The trackedTrade mtx must
+// NOT be locked.
+func (t *trackedTrade) auditContract(match *matchTracker, coinID []byte, contract []byte, timeout time.Duration) error {
 	// Get the asset.AuditInfo from the ExchangeWallet. Handle network latency.
 	// The coin waiter will run once every recheckInterval until successful or
-	// until expiration after broadcast timeout. The client is asked by the
-	// server to audit a contract transaction, and they have until broadcast
-	// timeout to do it before they get penalized and the match revoked. Thus,
-	// there is no reason to give up on the request sooner since the server will
-	// not ask again.
+	// until the match is revoked. The client is asked by the server to audit a
+	// contract transaction, and they have until broadcast timeout to do it
+	// before they get penalized and the match revoked. Thus, there is no reason
+	// to give up on the request sooner since the server will not ask again and
+	// the client will not solicit the counterparty contract data again except
+	// on reconnect.
 	errChan := make(chan error, 1)
 	var auditInfo asset.AuditInfo
+	var tries int
+	contractID, contractSymb := coinIDString(t.wallets.toAsset.ID, coinID), t.wallets.toAsset.Symbol
 	t.latencyQ.Wait(&wait.Waiter{
-		Expiration: time.Now().Add(t.broadcastTimeout()),
+		Expiration: time.Now().Add(timeout),
 		TryFunc: func() bool {
 			var err error
 			auditInfo, err = t.wallets.toWallet.AuditContract(coinID, contract)
-			if err != nil {
-				if err == asset.CoinNotFoundError {
-					return wait.TryAgain
-				}
-				errChan <- err
+			if err == nil {
+				// Success.
+				errChan <- nil
 				return wait.DontTryAgain
 			}
-			errChan <- nil
+			if errors.Is(err, asset.CoinNotFoundError) {
+				// Didn't find it that time.
+				if t.MatchIsRevoked(match) {
+					errChan <- ExpirationErr(fmt.Sprintf("match revoked while waiting to find counterparty contract coin %v (%s). "+
+						"Check your internet and wallet connections!", contractID, contractSymb))
+					return wait.DontTryAgain
+				}
+				if tries > 2 {
+					t.dc.log.Infof("Still searching for contract coin %v (%s) for match %v. "+
```

Comment:
Still considering if it should still try at least a couple times even if it is revoked, or is being revoked a good indicator that the audit now irrelevant...

Reply:
Audit is only useful for the normal swap negotiation path, so I've concluded no, the audit is not needed for a revoked match, which will go to refund.  However, I believe there is an opportunity for an improvement that would require more work: https://github.com/decred/dcrdex/issues/829